Stripe/STPPaymentContext.m

```diff
@@ -648,15 +653,25 @@ - (void)requestPayment {
                             if ([source isKindOfClass:[STPToken class]]) {
                                 paymentResultSource = ((STPToken *)source).card;
                             }
-                            STPPaymentResult *result = [[STPPaymentResult alloc] initWithSource:paymentResultSource];
+                            result = [[STPPaymentResult alloc] initWithSource:paymentResultSource];
                             [self.delegate paymentContext:self didCreatePaymentResult:result completion:^(NSError * error) {
                                 // for Apple Pay, the didFinishWithStatus callback is fired later when Apple Pay VC finishes
+                                // The apple pay vc finishes when the user cancels and it finishes only once.
+                                // So if the user has already canceled calling completion will not cause the apple pay vc to finish
+                                // Therefore, we must directly call our did finish with status method
```

Comment:
hmm, trying to understand this flow. Could you point out where `completion` is created?